hook.js

```diff
@@ -14,6 +14,14 @@ const NODE_MINOR = Number(NODE_VERSION[1])
 
 let entrypoint
 
+if (NODE_MAJOR >= 20) {
+  getExports = require('./lib/get-exports.js')
+} else {
+  getExports = async function (url) {
+    return Object.keys(await import(url))
+  }
```

Comment:
Could also be written like this to slightly reduce promise activity and be a bit more minimal, but it's probably fine as-is too.

```suggestion
  getExports = (url) => import(url).then(Object.keys)
```